lib/api-helper/project/fileCopy.ts

```diff
@@ -74,3 +78,31 @@ export function copyFiles(donorProject: Project,
         return p;
     };
 }
+
+/**
+ * Take the specified files from the donor project
+ * @param {RemoteRepoRef} donorProjectId
+ * @param {FileGlobMapping} fileGlobMapping - treated as globs as defined in Project.streamFiles
+ * @param {ProjectOperationCredentials} credentials
+ * @return {SimpleProjectEditor}
+ */
+export function streamFilesFrom(donorProjectId: RemoteRepoRef,
+                                fileGlobMapping: FileGlobMapping,
+                                credentials: ProjectOperationCredentials): CodeTransform {
+    return async (p, i) => {
+        const donorProject = await GitCommandGitProject.cloned(credentials, donorProjectId);
+        return streamFiles(donorProject, fileGlobMapping)(p, i);
+    };
+}
+
+export function streamFiles(donorProject: Project,
+                            fileGlobMapping: FileGlobMapping): CodeTransform {
+    logger.debug("fileGlobMappings: %s", fileGlobMapping);
+    return async p => {
+        donorProject.streamFilesRaw(fileGlobMapping.donorPath, { })
+            .on("data", f => {
+                logger.debug("file: %s", f);
+                p.addFileSync(fileGlobMapping.recipientPath + f.path, f.getContentSync());
+            });
```

Comment:
Wouldn't you also need an `.on("end", …)` to resolve the Promise? I would think you would need to return a `new Promise((resolve, reject) => { … });` and deal with the stream events. But I'm not an expert on Node.js streaming.

Reply:
Ok, I'll read up on nodejs streaming. Very new to this :-)

Reply:
My impression si that when working with file content streams, the `end` and `error` event is important in order to close the resources used during the stream. That is not the case here as we get distinct file objects per event.

Reply:
Here, you are receiving a stream of File objects, but they stream will end and will send that event when it does. Only then should the Promise be resolved. See  See https://stackoverflow.com/a/44013049 .

Reply:
https://nodejs.org/api/stream.html#stream_class_stream_readable says nothing about resolve. No method called resolve in all of nodejs stream doc.

According to https://github.com/gulpjs/glob-stream the stream used is a Readable stream.

In https://github.com/nodejs/node/blob/master/doc/api/stream.md#event-end it is said that you should consume the stream and it will send an end event when done. I don't believe I need to do anything special on end. I can of course throw in a debug statement so we see that it gets the end event?